plugin-server/tests/main/db.test.ts

```diff
@@ -817,6 +818,28 @@ describe('DB', () => {
         })
     })
 
+    describe('postgresQuery', () => {
+        it('caches query results if cacheResult=true', async () => {
+            jest.spyOn(db, 'redisSet')
+            const queryTag = 'testCachedQuery'
+            await db.postgresQuery('SELECT 1 as col', undefined, queryTag, undefined, true)
+            expect(db.redisSet).toHaveBeenCalledWith(
+                `${POSTGRES_QUERY_CACHE_PREFIX}${queryTag}`,
+                JSON.stringify({ rows: [{ col: 1 }], rowCount: 1 }),
+                expect.any(Number)
+            )
+        })
+
+        it('returns cached results if cacheResult=true', async () => {
+            const queryTag = 'testCachedQuery'
+            await db.postgresQuery('SELECT 1 as col', undefined, queryTag, undefined, true)
+            const res = await db.postgresQuery('SELECT 2 as col', undefined, queryTag, undefined, true)
+
+            // if this wasn't cached the value would have been 2
```

Comment:
Can we make it explicit with something like `expect(db.redisSet).not.toHaveBeenCalled()`?

Reply:
It was called - maybe the comment is more confusing than helpful. We receive a 1 back rather than a 2, meaning this was cached

Reply:
I think the comment was helpful

Reply:
`.toBeCalledOnce()` or equivalent to ensure that set was called on the first query only, but not on the second one?

Reply:
Ah I see now - @neilkakkar's comment helped me understand @guidoiaquinti's. I was confused at first because I thought you were asking me to assert that it was called. Neil's suggestion is exactly what we need here